packages/block-library/src/embed/edit.js

```diff
@@ -201,21 +221,18 @@ const EmbedEdit = ( props ) => {
 					icon={ icon }
 					label={ label }
 					onFocus={ onFocus }
-					onSubmit={ ( event ) => {
-						if ( event ) {
-							event.preventDefault();
-						}
-
-						setIsEditingURL( false );
-						setAttributes( { url } );
-					} }
+					onSubmit={ Platform.select( {
+						web: onSubmit,
+						native: onSubmitNative,
+					} ) }
 					value={ url }
 					cannotEmbed={ cannotEmbed }
 					onChange={ ( event ) => setURL( event.target.value ) }
 					fallback={ () => fallback( url, onReplace ) }
 					tryAgain={ () => {
 						invalidateResolution( 'getEmbedPreview', [ url ] );
 					} }
+					isSelected={ isSelected }
```

Comment:
This prop was added because it's required in the calculation for showing automatically the embed sheet ([code reference](https://github.com/WordPress/gutenberg/pull/31343/files#diff-794d5b4e5171d2fc3a36373cdbb2abd0fa4dd7a5892a759891ddaf1af0563accR33-R43)).